Compile/src/Producers/PUJetIDWeightProducer.cc

```diff
@@ -0,0 +1,85 @@
+#include "Excalibur/Compile/interface/Producers/PUJetIDWeightProducer.h"
+#include "KappaTools/RootTools/interface/Matching.h"
+
+std::string PUJetIDWeightProducer::GetProducerId() const { return "PUJetIDWeightProducer"; }
+
+void PUJetIDWeightProducer::Init(ZJetSettings const& settings) {
+
+    PUJetID = settings.GetPUJetID();
+    PUJetIDSelectionMaxPt = 50.0;
```

Comment:
Don't you want to have a setting for this? You could easily add a default value of `50.` there with:
```cpp
IMPL_SETTING_DEFAULT(std::float, PUJetIDSelectionMaxPt, 50.)
```

Reply:
Changed it to check whether an SF is available for a jet inside the range of the provided histogram.